testsuite/features/support/twopence_init.rb

```diff
@@ -334,19 +334,14 @@ def client_public_ip(host)
     return 'NOT_IMPLEMENTED' if node == it
   end
 
-  interface = case host
-              when /^deblike/, /^debian11/, /^ubuntu/
-                $is_cloud_provider ? 'eth0' : 'ens3'
-              when 'kvm_server', 'xen_server'
-                'br0'
-              else
-                'eth0'
-              end
-  node.init_public_interface(interface)
-  output, code = node.run("ip address show dev #{interface} | grep 'inet '")
-  raise 'Cannot resolve public ip' unless code.zero?
-
-  output.split[1].split('/')[0]
+  %w[br0 eth0 eth1 ens0 ens1 ens2 ens3 ens4 ens5 ens6].each do |dev|
```

Comment:
I have a doubt, if we run retail tests in AWS, we also need to adjust the next section where we set the private interface?

Reply:
Currently it seems that retail tests cannot be run in the cloud. So currently it does not create problems. But yes, potential we would need a detection as well.
But the detection in general is a bit weak. Just the lowest interface number is the "public" interface?

But currently it is the best we can do. Would it be ok, to use this for now and think about a better solution later?

Reply:
If Retail is out of the scope for AWS, then let's keep that improvement for later yes.
In any case, I think a card for this, to remind it will be good.

Reply:
https://github.com/SUSE/spacewalk/issues/19148